File: Problem_Solving/Algorithms/Warmup/3_Compare_The_Triplets/Solution.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

public class Solution {

    static List<Integer> compareTriplets(List<Integer> a, List<Integer> b) {
        List<Integer> results=new ArrayList<Integer>();
        int aliceScore=0;
        int bobScore=0;
        int i=0;
        int aliceCategory=0;
        int bobCategory=0;
        int numberOfCategories=a.size();

        do {
            aliceCategory=a.get(i);
            bobCategory=b.get(i);

            if(aliceCategory>bobCategory)
                aliceScore++;
            else if (aliceCategory<bobCategory)
                bobScore++;

            i++;
        }while(i<numberOfCategories);

        results.add(aliceScore);
        results.add(bobScore);
        return results;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        List<Integer> a = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        List<Integer> b = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        List<Integer> result = compareTriplets(a, b);

        bufferedWriter.write(
            result.stream()
                .map(Object::toString)
                .collect(joining(" "))
            + "\n"
        );

        bufferedReader.close();
        bufferedWriter.close();
    }
}

File: Problem_Solving/Algorithms/Warmup/3_Compare_The_Triplets/Solution.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

function compareTriplets(a, b) {
    const results=[];
    let aliceScore=0;
    let bobScore=0;
    let aliceCategory=0;
    let bobCategory=0;
    let numberOfCategories=a.length;

    for(let i=0;i<numberOfCategories;i++){
        aliceCategory=a[i];
        bobCategory=b[i];

        if(aliceCategory>bobCategory)
            aliceScore++;
        else if(aliceCategory<bobCategory)
            bobScore++;
    }

    results.push(aliceScore,bobScore);
    return results;
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const a = readLine().replace(/\s+$/g, '').split(' ').map(aTemp => parseInt(aTemp, 10));

    const b = readLine().replace(/\s+$/g, '').split(' ').map(bTemp => parseInt(bTemp, 10));

    const result = compareTriplets(a, b);

    ws.write(result.join(' ') + '\n');

    ws.end();
}
